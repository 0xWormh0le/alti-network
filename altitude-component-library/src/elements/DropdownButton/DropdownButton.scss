@import '../../styles/_shared.scss';

.DropdownWrapper {
  position: relative;
  justify-content: flex-start !important;

  .Button--dropdown {
    padding: 4px 20px 4px 12px;
    &:hover {
      polygon {
        fill: white;
      }
    }
  }

  &.disabled {
    svg {
      polygon {
        fill: #9fa9ba;
      }
    }

    &:focus-within {
      .DropdownPopup {
        display: none;
      }
    }
  }

  svg {
    width: 6px;
    margin-left: -14px;
    margin-top: 12px;
  }

  @include flex(row, center);

  .DropdownPopup {
    display: none;
    position: absolute;
    margin-top: 2.75em;
    width: 114px;
    height: auto;
    border-radius: 4px;
    box-shadow: 0 5px 15px 0 rgba(0, 0, 0, 0.07), 0 15px 35px 0 rgba(50, 50, 93, 0.1);
    padding: 0;
    z-index: 2;
    background: white;
    // text-transform: uppercase;

    &:hover {
      & > :first-child {
        background-color: $VIBRANT_BLUE;
      }
    }

    &--arrow {
      position: absolute;
      width: 6px;
      height: 6px;
      background: white;
      top: -2px;
      left: 54px;
      transform: rotate(45deg);
      z-index: -2;
    }

    &--arrow-off {
      polygon {
        fill: $DUSK_BLUE_DARK;
        transition: fill $TRANSITION_DURATION ease-in-out, color $TRANSITION_DURATION ease-in-out,
          border $TRANSITION_DURATION ease-in-out;
      }
    }

    &--arrow-on {
      polygon {
        fill: white;
        transition: fill $TRANSITION_DURATION ease-in-out, color $TRANSITION_DURATION ease-in-out,
          border $TRANSITION_DURATION ease-in-out;
      }
    }

    li {
      list-style-type: none;
      font-family: LatoLatin;
      font-size: 10px;
      font-weight: bold;
      font-stretch: normal;
      font-style: normal;
      line-height: normal;
      letter-spacing: 0.28px;
      line-height: 2.2;
      height: 30px;
      padding: 4px;
      // text-transform: uppercase;
      cursor: pointer;
      text-align: center;

      &:last-of-type:hover {
        border-bottom-left-radius: 4px;
        border-bottom-right-radius: 4px;
      }

      &:first-of-type {
        border-top-left-radius: 4px;
        border-top-right-radius: 4px;
      }

      a {
        color: #354052;
        text-decoration: none;
        display: block;
      }

      &.disabled {
        color: #9fa9ba;
        display: none;

        &:hover {
          background-color: white;
          color: #9fa9ba;
        }
      }

      &:hover {
        background-color: $VIBRANT_BLUE;
        color: white;
        a {
          color: white;
        }
      }
    }
  }

  &:focus-within {
    outline-style: none;

    .Button--dropdown {
      color: #9fa9ba;
      border-color: #9fa9ba;
    }

    svg {
      transform: rotate(180deg);

      polygon {
        fill: #9fa9ba;
      }
    }

    &.disabled {
      svg {
        transform: rotate(360deg);
      }
    }
  }

  &:focus-within .DropdownPopup {
    display: block;
  }
}
