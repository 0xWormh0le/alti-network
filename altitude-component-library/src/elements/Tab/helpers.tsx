export const getTabIndexOf = (element: HTMLElement) => {
  const tabs = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tab') as HTMLElement[]
  return tabs.indexOf(element)
}

export const getTabPanelIndexOf = (element: HTMLElement) => {
  const tabPanels = [].slice
    .call(element?.parentNode?.children)
    .filter((e: HTMLElement) => e.getAttribute('aria-roledescription') === 'tabPanel') as HTMLElement[]
  return tabPanels.indexOf(element)
}
