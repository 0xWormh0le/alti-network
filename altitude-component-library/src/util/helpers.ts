import { Location } from 'history'
import fp from 'lodash/fp'
import moment from 'moment'
import RiskCatalog from 'models/RiskCatalog'

const generateRoutePath = (label: string) => {
  // convert to small caps, replace spaces with dashes, and add forward slash at the beginning
  const lowerCaseLabel = label.toLowerCase().replace(' ', '-')
  return `/${lowerCaseLabel}`
}

export const toProperCase = (str: string) => {
  return str
    .split(' ')
    .map((item) => {
      const newString = item[0].toUpperCase()
      return newString + item.slice(1)
    })
    .join(' ')
}

export const getRiskText = (_riskId: string) => {
  const riskId = _riskId ? parseInt(_riskId, 10) : 0
  const risk = Object.keys(RiskCatalog).find(
    (item: string) => RiskCatalog[item].hasOwnProperty('index') && RiskCatalog[item].index === riskId
  )
  if (risk) {
    let riskText
    riskText = RiskCatalog[risk].name.replace(/Many Files/, 'Files')
    riskText = riskText.replace(/24 hours/, 'the past 24 hours')
    return riskText
  }
  return 'Files associated with this risk'
}

const capitalize = (input: string) => {
  if (input.length > 0) {
    const output = input.toLowerCase().split('')
    output[0] = output[0].toUpperCase()
    return output.join('')
  }
  return input
}

const ellipsify = (input: string, numberOfChars: number) => {
  if (!input) {
    return ''
  } else if (input.length > numberOfChars) {
    return input.substring(0, numberOfChars) + 'â€¦'
  } else {
    return input
  }
}

const getRisk = (value: SeverityRange) => {
  return {
    veryhigh: value > 9,
    high: value > 7,
    medium: value > 5 && value <= 7,
    low: value > 2 && value <= 5,
    safe: value > -1 && value <= 2
  }
}

const getRiskLabel = (value: SeverityRange) => {
  if (value > 9) {
    return 'Very High'
  }
  if (value > 7) {
    return 'High'
  }
  if (value > 5 && value <= 7) {
    return 'Medium'
  }
  if (value > 2 && value <= 5) {
    return 'Low'
  }
  if (value > -1 && value <= 2) {
    return 'Safe'
  }
  throw new Error(`Cannot label risk with severity ${value}`)
}

/* This test needs to cover 2 cases: 
  1. Value comes in as an immutable object
  2. Value comes in as a straight object
  If the value comes in as a straight object and sensitivePhrases.hasOwnProperty('sensitiveKeywords'), it will be
  an array, so we need to check length instead of size
*/
export const hasSensitiveContent = (value: any) => {
  const sensitivePhrases = toJs(value as IValue<ISensitiveObject>, 'sensitivePhrases') as ISensitiveObject
  if (sensitivePhrases.hasOwnProperty('ccFileNumCount') && sensitivePhrases.ccFileNumCount > 0) {
    return true
  } else if (
    sensitivePhrases.hasOwnProperty('sensitiveKeywordsFileCount') &&
    sensitivePhrases.sensitiveKeywordsFileCount > 0
  ) {
    return true
  } else if (sensitivePhrases.hasOwnProperty('ssnFileCount') && sensitivePhrases.ssnFileCount > 0) {
    return true
  } else if (sensitivePhrases.hasOwnProperty('ssn') && sensitivePhrases.ssn > 0) {
    return true
  } else if (sensitivePhrases.hasOwnProperty('ccNum') && sensitivePhrases.ccNum > 0) {
    return true
  } else if (sensitivePhrases.hasOwnProperty('sensitiveKeywords') && sensitivePhrases.sensitiveKeywords.size > 0) {
    return true
  } else if (
    sensitivePhrases.hasOwnProperty('sensitiveKeywords') &&
    sensitivePhrases.sensitiveKeywords.hasOwnProperty('length')
  ) {
    return sensitivePhrases.sensitiveKeywords.length > 0
  }
  return false
}

export const hasKeywords = (sensitivePhrases: ISensitivePhrase) => {
  return sensitivePhrases.hasOwnProperty('sensitiveKeywords') && sensitivePhrases.sensitiveKeywords!.size > 0
}

export const getKeywords = (sensitivePhrases: any) => {
  let additional = 0
  const phrases: IPhrase[] = []
  if (hasKeywords(sensitivePhrases)) {
    let total = 0
    sensitivePhrases.sensitiveKeywords.forEach((index: number) => {
      const phraseMap = sensitivePhrases.sensitiveKeywords.get(index)
      total++
      if (total < 6) {
        const count = phraseMap.get('count')
        const keyword = phraseMap.get('keyword')
        phrases.push({
          count,
          keyword
        })
      } else {
        additional += phraseMap.get('count')
      }
    })
  }
  return { phrases, additional }
}

const isMobileOrTablet = () => {
  const { userAgent } = window.navigator
  return /Mobile/i.test(userAgent) || /Android/i.test(userAgent)
}

interface IImmutable {
  get: (x: any) => any
}

export const toJs = (immutableObject: IImmutable, immutableString: string | number) => {
  if (typeof immutableObject.get === 'undefined') {
    return immutableObject.hasOwnProperty(immutableString) ? immutableObject[immutableString] : immutableObject
  }
  const immutableMap = immutableObject.get(immutableString)
  const retval = {}
  if (immutableMap && typeof immutableMap.forEach === 'function') {
    immutableMap.forEach((value: any, key: any) => {
      retval[key] = value
    })
  }
  return retval
}

const modalRoutes = [
  '/spotlight/',
  '/file/',
  '/files/',
  '/people/',
  '/app-spotlight/',
  '/permissions/file/',
  '/resolve/'
]

const locationIsAModalRoute = (location: string) => modalRoutes.some((modalRoute) => location.includes(modalRoute))
const searchWithoutModalParams = (extraParams = {}) => {
  // This function removes modal-specific query params between routes. Passing other
  // params during modal navigation (navigating to or between modals) retains state,
  // which is desired. But the params for modal state cannot be passed between modals
  // since modal state should be reset on modal navigation.
  // `extraParams` is appended to the query string if provided.
  const searchString = window.location.search
  const finalString = searchString
    .replace(/[&?]modalPage=[0-9]+/g, '')
    .replace(/[&?]spotlightCard=[a-z]+/g, '')
    .replace(/[&?]selectedEmail=[\w+.%]+/g, '')
    .replace(/[&?]breadcrumbPosition=[0-9]+/g, '')
    .replace(/[&?]selectedSubNavKey=[a-z]+/g, '')
    .replace(/[&?]utm-source=([a-zA-Z'-]+)/g, '') // match parameters like &utm-source=risk-summary-email

  const queryParams = {
    ...parseQueryString(finalString),
    ...extraParams
  }
  return jsonToQueryString(queryParams)
}

export const getExtraParams = (riskTypeId: RiskTypeId) => {
  const isManyDownloadsByPerson =
    RiskCatalog.ManyDownloadsByPersonInternal.index === riskTypeId ||
    RiskCatalog.ManyDownloadsByPersonExternal.index === riskTypeId
  const extraParams = isManyDownloadsByPerson ? { selectedSubNavKey: 'personDownloads' } : {}
  return searchWithoutModalParams(extraParams)
}

const replaceUrlParam = (url: string, paramName: string, paramValue: string) => {
  if (paramValue == null) {
    paramValue = ''
  }
  const pattern = new RegExp('\\b(' + paramName + '=).*?(&|#|$)')
  if (url.search(pattern) >= 0) {
    return url.replace(pattern, '$1' + paramValue + '$2')
  }
  url = url.replace(/[?#]$/, '')
  return url + (url.indexOf('?') > 0 ? '&' : '?') + paramName + '=' + paramValue
}

const largeNumberDisplay = (n: number): string => {
  const LOWER_THRESHOLD = 10000
  const UPPER_THRESHOLD = 1000000

  if (n >= LOWER_THRESHOLD && n < UPPER_THRESHOLD) {
    return `${Math.floor(n / 1000)}k+`
  } else if (n >= UPPER_THRESHOLD) {
    return `${Math.floor(n / 1000000)}m+`
  }

  return n.toString()
}

const largeNumberWithPluralizedUnit = (largeNumber: number, unit: string) =>
  `${largeNumberDisplay(largeNumber)} ${pluralize(unit, largeNumber)}`

const pluralize = (txt: string, cnt: number) => {
  if (cnt === 1) {
    return txt
  }

  const plural = {
    '(quiz)$': '$1zes',
    '^(ox)$': '$1en',
    '([m|l])ouse$': '$1ice',
    '(matr|vert|ind)ix|ex$': '$1ices',
    '(x|ch|ss|sh)$': '$1es',
    '([^aeiouy]|qu)y$': '$1ies',
    '(hive)$': '$1s',
    '(?:([^f])fe|([lr])f)$': '$1$2ves',
    '(shea|lea|loa|thie)f$': '$1ves',
    sis$: 'ses',
    '([ti])um$': '$1a',
    '(tomat|potat|ech|her|vet)o$': '$1oes',
    '(bu)s$': '$1ses',
    '(alias)$': '$1es',
    '(octop)us$': '$1i',
    '(ax|test)is$': '$1es',
    '(us)$': '$1es',
    '([^s]+)$': '$1s'
  }

  const irregular = {
    move: 'moves',
    foot: 'feet',
    goose: 'geese',
    sex: 'sexes',
    child: 'children',
    man: 'men',
    tooth: 'teeth',
    person: 'people'
  }

  const uncountable = [
    'sheep',
    'fish',
    'deer',
    'moose',
    'series',
    'species',
    'money',
    'rice',
    'information',
    'equipment'
  ]

  // save some time in the case that singular and plural are the same
  if (uncountable.indexOf(txt.toLowerCase()) >= 0) {
    return txt
  }

  // check for irregular forms
  for (const word in irregular) {
    if (irregular.hasOwnProperty(word)) {
      const pattern = new RegExp(word + '$', 'i')
      const replace = irregular[word]

      if (pattern.test(txt)) {
        return txt.replace(pattern, replace)
      }
    }
  }

  const array = plural

  // check for matches using regular expressions
  for (const reg in array) {
    if (array.hasOwnProperty(reg)) {
      const ptn = new RegExp(reg, 'i')

      if (ptn.test(txt)) {
        return txt.replace(ptn, array[reg])
      }
    }
  }

  return txt
}

export const copyToClipboard = (txt: string) => {
  const element = document.createElement('textarea')
  element.value = txt
  element.style.position = 'absolute'
  element.style.left = '-9999px'
  document.body.appendChild(element)
  element.select()
  document.execCommand('copy')
  document.body.removeChild(element)
}

export const truncateText = (txt: string, maxlen: number = 20) => {
  if (txt.length > maxlen) {
    return txt.substr(0, maxlen) + '...'
  }
  return txt
}

const addCSVColumn = (content?: string) => {
  return `${content || ''},`
}

export const isDemoEnv = ['demo', 'localdemo'].includes(process.env.REACT_APP_STAGE || '')
export const isDevEnv = ['dev'].includes(process.env.REACT_APP_STAGE || '')

export const getLoginRedirectPath = (location: Location<any>): string => {
  let nextPath = location.pathname
  if (location.search) {
    nextPath += location.search.replace('?', '&')
  }
  return `/login?nextPath=${nextPath}`
}

export const parseQueryString = (query: string): { [key: string]: string } =>
  fp.compose(
    JSON.parse,
    JSON.stringify,
    fp.reduce((acc: object, part: string) => {
      const [name, value] = part.split('=')
      acc[name] = decodeURIComponent(value || '')
      return acc
    }, {}),
    (str) => (str ? str.split('&') : []),
    fp.replace('?', '')
  )(query)

export const jsonToQueryString = (queryParams: { [key: string]: string | number | boolean }): string => {
  const pairs: string[] = []
  if (queryParams) {
    Object.keys(queryParams).forEach((key) => {
      if (queryParams[key] !== null && typeof queryParams[key] !== 'undefined') {
        const value = encodeURIComponent(queryParams[key].toString())
        if (value) {
          pairs.push(`${key}=${value}`)
        }
      }
    })
  }
  return pairs.length ? `?${pairs.join('&')}` : ''
}

// tslint:disable-next-line no-empty
export const noop = () => {}

export const getEmailDomain = (email: string): string => email.replace(/.*@/, '')

export const isExternalEmail = (domains: string[] = [], creatorEmail: string): boolean => {
  return Boolean(creatorEmail) && !domains.includes(getEmailDomain(creatorEmail))
}

export const getFullName = ({ firstName, lastName }: { firstName?: string; lastName?: string }): string =>
  fp
    .reject((item) => !item)([firstName, lastName])
    .join(' ')

export const findMatches = (text: string, keyword: string): number[][] => {
  const regex = new RegExp(keyword, 'gi')
  const matches = []
  let match = regex.exec(text)
  while (match !== null) {
    matches.push([match.index, match.index + keyword.length])
    match = regex.exec(text)
  }
  return matches
}

export const filterWithLimit = <T>(collection: T[], predicate: (item: T) => boolean, limit: number): T[] => {
  const results: T[] = []
  fp.forEach((item: T): boolean | undefined => {
    if (results.length === limit) {
      return false
    }
    if (predicate(item)) {
      results.push(item)
    }
    return undefined
  })(collection)
  return results
}

export const joinCSVCells = (cells: string | string[]): string => (Array.isArray(cells) ? cells.join(',') : cells)

export const printCSVdate = (unixTime: number, format: string, utc?: boolean) => DateUtils.unix(unixTime, format, utc)

export class DateUtils {
  static unix(unixTime: number, format: string, utc?: boolean) {
    if (utc) {
      return moment.unix(unixTime).utc().format(format)
    }
    return moment.unix(unixTime).format(format)
  }

  static getMoment(dateValue: number | string, utc: boolean = false) {
    const self = new DateUtils()
    return self._parsedDate(dateValue, utc)
  }

  _parsedDate(dateValue: number | string, utc: boolean) {
    let parsedDate
    if (isNumber(dateValue)) {
      parsedDate = utc ? moment.unix(dateValue as number).utc() : moment.unix(dateValue as number)
    } else {
      parsedDate = moment(dateValue)
    }
    return parsedDate
  }

  static now(format: string) {
    return moment().format(format)
  }

  static timestamp(value?: string | number, utc?: boolean) {
    if (value) {
      return utc ? moment.utc(value).unix() : moment(value).unix()
    } else {
      return utc ? moment().utc().unix : moment().unix()
    }
  }

  static dateFormat(dateValue: string | number, format: string, utc?: boolean) {
    const self = new DateUtils()
    const _moment = self._parsedDate(dateValue, utc || false)
    return _moment.format(format)
  }
}

export const csvFileName = (fileName: string, format: string) => {
  return `${fileName}_${DateUtils.dateFormat(DateUtils.now(format), 'DD-MM-YYYY_HH-mm-ss')}.csv`
}

export const isNumber = (value: any) => {
  return typeof value === 'number' && isFinite(value)
}

export {
  generateRoutePath,
  getRisk,
  getRiskLabel,
  capitalize,
  ellipsify,
  isMobileOrTablet,
  modalRoutes,
  locationIsAModalRoute,
  searchWithoutModalParams,
  largeNumberDisplay,
  pluralize,
  largeNumberWithPluralizedUnit,
  addCSVColumn,
  replaceUrlParam
}

export function getQueryString(name: string) {
  const url = window.location.href
  name = name.replace(/[[]]/g, '\\$&')
  const regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)')
  const results = regex.exec(url)
  if (!results) return null
  if (!results[2]) return ''
  return decodeURIComponent(results[2].replace(/\+/g, ' '))
}
