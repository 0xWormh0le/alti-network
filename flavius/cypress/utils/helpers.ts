export const capitalize = (input: string) => {
  if (input.length > 0) {
    const output = input.toLowerCase().split('')
    output[0] = output[0].toUpperCase()
    return output.join('')
  }
  return input
}
