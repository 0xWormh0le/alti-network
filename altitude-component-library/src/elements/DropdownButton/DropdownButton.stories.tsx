import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DropdownButton, { DropdownButtonProps } from './DropdownButton'

export default {
  title: 'Elements/DropdownButton',
  component: DropdownButton
} as Meta

const Template: Story<DropdownButtonProps> = (args) => <DropdownButton {...args} />

export const Default = Template.bind({})
Default.args = {
  text: 'Actions Header',
  enabled: true,
  actions: [
    {
      title: 'Enabled Action1',
      actionEnabled: true,
      link: '',
      onClick: () => null
    },
    {
      title: 'Disabled Action',
      actionEnabled: false,
      link: '',
      onClick: () => null
    }
  ]
}

export const Disabled = Template.bind({})
Disabled.args = {
  enabled: false,
  text: 'Actions Header',
  actions: []
}
