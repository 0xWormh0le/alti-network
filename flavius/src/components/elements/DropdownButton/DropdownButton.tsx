import React, { useState } from 'react'
import Button from '../Button'
import cx from 'classnames'
import { Link } from 'react-router-dom'

import './DropdownButton.scss'

const Arrow: React.FC<{ arrowClass: string }> = ({ arrowClass }) => {
  return (
    <svg width='12px' height='8px' viewBox='0 0 12 8' version='1.1' className={arrowClass}>
      <polygon id='path-1' points='6 8 12 0 0 0' />
    </svg>
  )
}

export interface ActionType {
  onClick: () => void
  title: string
  link: any
  actionEnabled: boolean
  actionSecondaryText?: string
}

export interface DropdownButtonProps {
  text: string
  actions: ActionType[]
  enabled: boolean
}

const DropdownButton: React.FC<DropdownButtonProps> = (props) => {
  const { text, actions, enabled } = props
  const [arrowClass, setArrowClass] = useState('DropdownPopup--arrow-off')

  return (
    <div className={cx('DropdownWrapper', { disabled: !enabled })} tabIndex={1}>
      <Button
        text={text}
        action='secondary'
        className={cx('Button--dropdown Button--size-medium', enabled ? '' : 'Button--disabled')}
        onMouseOver={() => {
          setArrowClass('DropdownPopup--arrow-on')
        }}
        onMouseOut={() => {
          setArrowClass('DropdownPopup--arrow-off')
        }}
      />
      <Arrow arrowClass={arrowClass} />
      <ul className='DropdownPopup'>
        <span className='DropdownPopup--arrow' />
        {actions.map(({ onClick, title, link, actionEnabled }, index) => {
          return actionEnabled ? (
            <li key={index} className={cx('DropdownPopup--item', { disabled: !actionEnabled })} onClick={onClick}>
              {link === '' ? <span>{title}</span> : <Link to={link}>{title}</Link>}
            </li>
          ) : (
            ''
          )
        })}
      </ul>
    </div>
  )
}

export default DropdownButton
