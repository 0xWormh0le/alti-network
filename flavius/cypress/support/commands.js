import * as Sentry from '@sentry/browser'
import { RISK_TYPE_IDS } from '../utils/config'
const { MANY_DOWNLOADS_BY_APP_ID } = RISK_TYPE_IDS
import { version } from '../../package.json'

Cypress.on('fail', (error, runnable) => {
  console.info('we failed!!', error, runnable)
  const errorMessage = `Heartbeat Error detected in spec '${runnable.parent.title}': ${error.message}`
  Sentry.init({
    dsn: 'https://4345e62a3abc48049f9965ccc17ed310@sentry.io/1306610',
    environment: Cypress.env('REACT_APP_STAGE'),
    release: `flavius@${version}`,
    integrations: [
      new Sentry.Integrations.Breadcrumbs({
        console: Cypress.env('REACT_APP_STAGE') === 'production',
        message: errorMessage
      })
    ]
  })
  const err = new Error(errorMessage)
  Sentry.withScope((scope) => {
    scope.setExtra('stacktrace', error.stack)
    scope.setExtra('codeframe', error.codeFrame)
    Sentry.captureException(err)
  })
  throw err
})

const checkMinimumValidPerson = (person) => {
  if (person) {
    expect(person.accessCount).to.be.a('number')
    expect(person.primaryEmail?.address).to.be.a('string')
    expect(person.hasOwnProperty('name')).to.be.true
  }
}

const checkMinimumFile = (file) => {
  const {
    createdAt,
    createdBy,
    externalAccessCount,
    externalAccessList,
    fileId,
    fileName,
    internalAccessCount,
    internalAccessList,
    lastModified,
    linkVisibility,
    lastIngested,
    iconUrl
  } = file
  if (createdAt) {
    expect(createdAt).to.be.a('number')
  }
  if (iconUrl) {
    expect(iconUrl).to.be.a('string')
  }
  expect(externalAccessCount).to.be.a('number')
  expect(externalAccessList).to.be.an('Array')
  // todo: uncomment. id and name is currently null
  expect(fileId).to.be.a('string')
  expect(fileName).to.be.a('string')
  expect(internalAccessCount).to.be.a('number')
  expect(internalAccessList).to.be.an('Array')
  if (lastModified) {
    expect(lastModified).to.be.a('number')
  }
  if (lastIngested) {
    expect(lastIngested).to.be.a('number')
  }
  expect(linkVisibility).to.be.a('string')
}

const checkMinimumFiles = (files) => checkMinimumFile(files[randomIndex(files)])

const randomIndex = (array) => Math.floor(Math.random() * array.length)

Cypress.Commands.add('login', () => {
  cy.get('#email').type('bobbie@thoughtlabs.io')
  cy.get('#password').type(Cypress.env('CYPRESS_USER_PASSWORD'))
  cy.get('.Login__button').click()
})

Cypress.Commands.add('checkMinimumValidPerson', checkMinimumValidPerson)

Cypress.Commands.add('checkFullPerson', (person) => {
  checkMinimumValidPerson(person)
  expect(person.externalCount).to.be.a('number')
  expect(person.userKind).to.be.a('number')
  expect(person.internalCount).to.be.a('number')
})

Cypress.Commands.add('checkFiles', (files) => {
  expect(files).to.be.an('Array')
  checkMinimumFiles(files)
})

Cypress.Commands.add('checkRisks', (risks, filteredRiskTypeId) => {
  expect(risks).to.be.an('Array')
  const randomRiskIndex = randomIndex(risks)
  const randomTestRisk = risks[randomRiskIndex]
  expect(randomTestRisk.platformId).to.be.a('string')
  expect(randomTestRisk.datetime).to.be.a('number')
  expect(randomTestRisk.fileCount).to.be.a('number')
  expect(randomTestRisk.hasOwnProperty('fileId')).to.be.true
  expect(randomTestRisk.hasOwnProperty('fileName')).to.be.true
  expect(randomTestRisk.riskDescription).to.be.a('string')
  expect(randomTestRisk.riskId).to.be.a('string')
  expect(randomTestRisk.riskTarget).to.be.an('Array')
  if (filteredRiskTypeId) {
    expect(randomTestRisk.riskTypeId).to.equal(filteredRiskTypeId)
  }
  expect(randomTestRisk.severity).to.be.a('number')

  checkMinimumValidPerson(randomTestRisk.creator)
  checkMinimumValidPerson(randomTestRisk.owner)

  if (filteredRiskTypeId === MANY_DOWNLOADS_BY_APP_ID) {
    expect(randomTestRisk.plugin.id).to.be.a('string')
    // todo: uncomment. name is currently null
    expect(randomTestRisk.plugin.name).to.be.a('string')
  }
})

Cypress.Commands.add('checkPagination', (paginationResponse, overrides, extra) => {
  expect(paginationResponse.orderBy).to.equal(
    overrides && overrides.hasOwnProperty('orderBy') ? overrides.orderBy : 'datetime'
  )
  expect(paginationResponse.pageCount).to.be.a('number')
  expect(paginationResponse.pageNumber).to.be.a('number')
  expect(paginationResponse.pageSize).to.be.a('number')
  expect(paginationResponse.sort).to.equal(overrides && overrides.hasOwnProperty('sort') ? overrides.sort : 'DESC')

  if (extra) {
    const { wrapper } = extra
    const { pageCount } = paginationResponse
    if (pageCount > 1) {
      if (pageCount > 5) {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', 5)
      } else {
        cy.get(wrapper + ' .PaginationItem--number').should('have.length', pageCount)
      }
    }
  }
})

Cypress.Commands.add('checkEvents', (events, filesIsObject) => {
  expect(events).to.be.an('Array')
  const randomEventIndex = randomIndex(events)
  const randomTestEvent = events[randomEventIndex]
  expect(randomTestEvent.datetime).to.be.a('number')
  expect(randomTestEvent.eventDescription).to.be.a('string')
  expect(randomTestEvent.eventId).to.be.a('string')
  expect(randomTestEvent.eventName).to.be.a('string')
  expect(randomTestEvent.targetPeople).to.be.an('Array')
  if (!filesIsObject) {
    expect(randomTestEvent.files).to.be.an('Array')
    const randomFileIndex = randomIndex(randomTestEvent.files)
    checkMinimumFile(randomTestEvent.files[randomFileIndex])
  } else {
    checkMinimumFiles(randomTestEvent.files)
  }
})
