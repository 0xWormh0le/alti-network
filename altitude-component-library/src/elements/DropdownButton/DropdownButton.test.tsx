import React from 'react'
import DropdownButton, { ActionType, DropdownButtonProps } from './DropdownButton'
import { noop, renderWithRouter } from 'test/support/helpers'

describe('DonutChart', () => {
  it('renders correctly', () => {
    const action: ActionType = {
      onClick: noop,
      title: 'add',
      link: 'https://google.com',
      actionEnabled: true,
    }

    const props: DropdownButtonProps = {
      text: 'dropdown button',
      actions: [action],
      enabled: false,
    }
    const { container } = renderWithRouter(<DropdownButton {...props} />)
    expect(container).toMatchSnapshot()
  })
})
