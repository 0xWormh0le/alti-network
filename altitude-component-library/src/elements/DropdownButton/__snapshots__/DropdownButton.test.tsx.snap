// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart renders correctly 1`] = `
<div>
  <div
    class="DropdownWrapper disabled"
    tabindex="1"
  >
    <button
      class="Button Button--dropdown Button--size-medium Button--disabled Button--secondary"
      type="button"
    >
      dropdown button
    </button>
    <svg
      class="DropdownPopup--arrow-off"
      height="8px"
      version="1.1"
      viewBox="0 0 12 8"
      width="12px"
    >
      <polygon
        id="path-1"
        points="6 8 12 0 0 0"
      />
    </svg>
    <ul
      class="DropdownPopup"
    >
      <span
        class="DropdownPopup--arrow"
      />
      <li
        class="DropdownPopup--item"
      >
        <a
          href="https://google.com"
        >
          add
        </a>
      </li>
    </ul>
  </div>
</div>
`;
